test/integration/node-middleware.test.ts

```diff
@@ -0,0 +1,90 @@
+import { createServer } from "http";
+
+import fetch from "node-fetch";
+
+import { Webhooks, createNodeMiddleware, sign } from "../../src";
+import { EmitterWebhookEvent } from "../../src/types";
+import { pushEventPayload } from "../fixtures";
+
+const signatureSha256 = sign(
+  { secret: "mySecret", algorithm: "sha256" },
+  JSON.stringify(pushEventPayload)
+);
+
+describe("createNodeMiddleware(webhooks)", () => {
+  it("README example", async () => {
+    const webhooks = new Webhooks<EmitterWebhookEvent>({
+      secret: "mySecret",
+    });
+
+    webhooks.on("push", (event) => {
+      expect(event.id).toBe("123e4567-e89b-12d3-a456-426655440000");
```

Comment:
because this is async, ideally you should use the optional `done` callback or (preferably) create a promise that gets resolved in this event, which you can `await` at the end of the test body.

Reply:
I'm not sure if I follow you here. The event handler is guaranteed to be executed before sending a response, as long as it doesn't take more than 9s, so the code should be executed before the `await fetch` below resolves